query-engine/prisma-models/src/field_selection.rs

```diff
@@ -245,6 +245,14 @@ impl From<Vec<Field>> for FieldSelection {
     }
 }
 
+impl From<Vec<ScalarFieldRef>> for FieldSelection {
+    fn from(fields: Vec<ScalarFieldRef>) -> Self {
+        Self {
+            selections: fields.into_iter().map(|sf| sf.into()).collect(),
```

Comment:
A neat shorthand is `fields.into_iter().map(Into::into).collect()`

Reply:
I feel like this could be generalized to `impl From<Vec<T>> for FieldSelection where T: Into<SelectedField>`, but I wonder why I didn't include this in the first place (maybe it doesn't work).